crates/client/src/lib.rs

```diff
@@ -246,16 +237,21 @@ impl<R: RegistryStorage, C: ContentStorage> Client<R, C> {
 
     /// Inserts or updates the logs of the specified packages in client storage to
     /// the latest registry checkpoint.
-    pub async fn upsert(&self, packages: &[&str]) -> Result<(), ClientError> {
+    pub async fn upsert<'a, I>(&self, packages: I) -> Result<(), ClientError>
```

Comment:
Do we use this in non-slice contexts now?

Reply:
We do in the tests currently. As this method is only iterating on the package ids with no requirement it needs to be a slice, it actually helps consumers like `cargo-component` that are iterating on some other data structure and projecting into it to get the warg package identifiers; this change removes the need for callers to collect into a contiguous collection.